src/bittybuzz/src/BittyBuzzMathFunctions.cpp

```diff
@@ -0,0 +1,311 @@
+#include "BittyBuzzMathFunctions.h"
+#include "BittyBuzzSystem.h"
+#include <bbzvm.h>
+#include <bsp/Math.h>
+
+void BittyBuzzMathFunctions::bbzmath_abs() {
+    bbzvm_assert_lnum(1); // NOLINT
+
+    // Get args
+    bbzobj_t* o = bbzheap_obj_at(bbzvm_locals_at(1)); // NOLINT
+    if (bbztype_isfloat(*o)) {
+        bbzvm_pushf(bbzfloat_fromfloat(Math::fabs(bbzfloat_tofloat(o->f.value))));
+    } else if (bbztype_isint(*o)) {
+        bbzvm_pushi(Math::abs(o->i.value));
+    } else {
+
+        bbzvm_seterror(BBZVM_ERROR_MATH);
+        return;
+    }
+    bbzvm_ret1();
+}
+void BittyBuzzMathFunctions::bbzmath_floor() {
+    bbzvm_assert_lnum(1); // NOLINT
+
+    // Get args
+    float arg = 0;
+    bbzobj_t* o = bbzheap_obj_at(bbzvm_locals_at(1)); // NOLINT
+    if (bbztype_isfloat(*o)) {
+        arg = bbzfloat_tofloat(o->f.value);
+    } else if (bbztype_isint(*o)) {
+        arg = o->i.value;
+    } else {
+        bbzvm_seterror(BBZVM_ERROR_MATH);
+        return;
```

Comment:
It seems that you repeat this code section in each math function expect the abs one. Is there a reason why you didn't create a function getArg() instead?

Reply:
Done